static/app/components/alerts/snoozeAlert.tsx

```diff
@@ -141,22 +154,31 @@ function SnoozeAlert({isSnoozed, onSnooze, projectSlug, ruleId, hasAccess}: Prop
         size="sm"
         icon={<IconSound />}
         disabled={disabled || !hasAccess}
-        onClick={() => handleMute('me')}
+        hasDropdown={ruleActionCategory !== RuleActionsCategories.NoDefault}
+        onClick={() => {
+          const target =
+            ruleActionCategory !== RuleActionsCategories.AllDefault ? 'everyone' : 'me';
+          handleMute(target);
+        }}
       >
-        {t('Mute')}
+        {ruleActionCategory === RuleActionsCategories.AllDefault
```

Comment:
Is there a reason why this logic is slightly different from the one in the `onClick`? Would make me feel better if they both shared logic so they can't diverge, maybe earlier do something like 

`const primaryMuteAction = RuleActionsCategories.AllDefault ? 'me' : 'everyone'`

Could even simplify a little more and do

`const primaryMuteAction = dropdownItems.find(item => !item.hidden)`

Reply:
no specific reason, i just flipped the check but having shared logic makes sense. i think having it go off the dropdown might make it a little more confusing though? especially since if there are no default integrations then there are no dropdown items since both are hidden.

Reply:
Ah yeah that's fair, just tossing out ideas that may be worse 😄